diff --git a/test/auto_close_test.cpp b/test/auto_close_test.cpp
index fb68022..897abfb 100644
--- a/test/auto_close_test.cpp
+++ b/test/auto_close_test.cpp
@@ -5,8 +5,11 @@
 
 // See http://www.boost.org/libs/iostreams for documentation.
 
+#define system invalid_system
+#include <stdlib.h>
 #include <cstdio>  // EOF.
 #include <boost/iostreams/filtering_stream.hpp>
+#undef system
 #include <boost/iostreams/stream.hpp>
 #include <boost/shared_ptr.hpp>
 #include <boost/test/test_tools.hpp>
