diff --git a/src/tools/gcc.jam b/src/tools/gcc.jam
index e5f8f53a4..66a066e53 100644
--- a/src/tools/gcc.jam
+++ b/src/tools/gcc.jam
@@ -397,7 +397,9 @@ local rule compile-link-flags ( * )
     # The default system libraries include pthread functions.  No additional libraries or CFLAGS are necessary to use this API.
     local no-threading = android beos haiku sgi darwin vxworks iphone appletv ;
     local threading-generic-os = [ set.difference $(all-os) : $(no-threading) $(bsd) windows cygwin solaris qnx ] ;
-    threading-flags <target-os>$(threading-generic-os) : -pthread : rt ;
+    #threading-flags <target-os>$(threading-generic-os) : -pthread : rt ;
+    # z/OS does not need -pthread
+    #threading-flags <target-os>$(threading-generic-os) : -pthread : rt ;
 }
 
 {
@@ -870,7 +872,7 @@ toolset.flags gcc.link OPTIONS <coverage>on : --coverage ;
     # at all and darwin.jam will use a separate 'strip' invocation.
     toolset.flags gcc.link RPATH <target-os>darwin : <dll-path> ;
     # This does not support -R.
-    toolset.flags gcc.link RPATH_OPTION <target-os>darwin : -rpath ;
+    #toolset.flags gcc.link RPATH_OPTION <target-os>darwin : -rpath ;
     # -rpath-link is not supported at all.
     toolset.flags gcc.link SHARED_OPTION <target-os>darwin : -dynamiclib -single_module ;
     toolset.flags gcc.link HAVE_SONAME <target-os>darwin : "" ;
@@ -897,13 +899,14 @@ toolset.flags gcc.link OPTIONS <coverage>on : --coverage ;
     # support -s.
     toolset.flags gcc.link OPTIONS <target-os>$(generic-os)/<strip>on :
         -Wl,--strip-all ;
-    toolset.flags gcc.link START-GROUP <target-os>$(generic-os) :
-        -Wl,--start-group ;
-    toolset.flags gcc.link END-GROUP <target-os>$(generic-os) : -Wl,--end-group ;
+# z/OS does not need start-/end-group
+    #toolset.flags gcc.link START-GROUP <target-os>$(generic-os) :
+     #   -Wl,--start-group ;
+    #toolset.flags gcc.link END-GROUP <target-os>$(generic-os) : -Wl,--end-group ;
 
     local rpath-os = [ set.difference $(all-os) : aix darwin vxworks solaris osf hpux windows ] ;
     toolset.flags gcc.link RPATH <target-os>$(rpath-os) : <dll-path> ;
-    toolset.flags gcc.link RPATH_OPTION <target-os>$(rpath-os) : -rpath ;
+    #toolset.flags gcc.link RPATH_OPTION <target-os>$(rpath-os) : -rpath ;
     toolset.flags gcc.link RPATH_LINK <target-os>$(rpath-os) : <xdll-path> ;
 
     # gnu ld has the ability to change the search behaviour for libraries
@@ -939,8 +942,9 @@ toolset.flags gcc.link OPTIONS <coverage>on : --coverage ;
     # search patterns!
 
     # On *nix mixing shared libs with static runtime is not a good idea.
-    toolset.flags gcc.link FINDLIBS-ST-PFX <target-os>$(generic-os)/<runtime-link>shared : -Wl,-Bstatic ;
-    toolset.flags gcc.link FINDLIBS-SA-PFX <target-os>$(generic-os)/<runtime-link>shared : -Wl,-Bdynamic ;
+    # # z/OS does not do -Bstatic/dynamic
+    #toolset.flags gcc.link FINDLIBS-ST-PFX <target-os>$(generic-os)/<runtime-link>shared : -Wl,-Bstatic ;
+    #toolset.flags gcc.link FINDLIBS-SA-PFX <target-os>$(generic-os)/<runtime-link>shared : -Wl,-Bdynamic ;
 
     # On windows allow mixing of static and dynamic libs with static
     # runtime is not a good idea.
@@ -950,7 +954,8 @@ toolset.flags gcc.link OPTIONS <coverage>on : --coverage ;
 
     local soname-os = [ set.difference $(generic-os) : windows ] ;
     toolset.flags gcc.link HAVE_SONAME <target-os>$(soname-os) : "" ;
-    toolset.flags gcc.link SONAME_OPTION <target-os>$(soname-os) : -soname ;
+    #toolset.flags gcc.link SONAME_OPTION <target-os>$(soname-os) : "" ;
+    #toolset.flags gcc.link SONAME_OPTION <target-os>$(soname-os) : -soname ;
 
     # See note [1]
     toolset.flags gcc.link OPTIONS <target-os>$(generic-os)/<runtime-link>static : -static ;
@@ -968,7 +973,7 @@ toolset.flags gcc.link OPTIONS <coverage>on : --coverage ;
     toolset.flags gcc.link OPTIONS <target-os>osf/<strip>on : -Wl,-s ;
     toolset.flags gcc.link RPATH <target-os>osf : <dll-path> ;
     # This does not support -R.
-    toolset.flags gcc.link RPATH_OPTION <target-os>osf : -rpath ;
+    #toolset.flags gcc.link RPATH_OPTION <target-os>osf : -rpath ;
     # -rpath-link is not supported at all.
 
     # See note [1]
@@ -1033,7 +1038,8 @@ rule link.dll ( targets * : sources * : properties * )
 
 actions link bind LIBRARIES
 {
-    "$(CONFIG_COMMAND)" @($(<[1]:T).rsp:O=FC:<=@":>=":E=-L"$(LINKPATH)" -Wl,$(RPATH_OPTION)$(SPACE)-Wl,$(RPATH) -Wl,-rpath-link$(SPACE)-Wl,"$(RPATH_LINK)" -o "$(<:T)" $(START-GROUP) "$(>:T)" "$(LIBRARIES)" $(FINDLIBS-ST-PFX) -l$(FINDLIBS-ST) $(FINDLIBS-SA-PFX) -l$(FINDLIBS-SA) $(END-GROUP) $(OPTIONS) $(USER_OPTIONS))
+    "$(CONFIG_COMMAND)" @($(<[1]:T).rsp:O=FC:<=@":>=":E=-L"$(LINKPATH)" -Wl,$(RPATH_OPTION)$(SPACE)-Wl,$(RPATH) -Wl, -o "$(<:T)" $(START-GROUP) "$(>:T)" "$(LIBRARIES)" $(FINDLIBS-ST-PFX) -l$(FINDLIBS-ST) $(FINDLIBS-SA-PFX) -l$(FINDLIBS-SA) $(END-GROUP) $(OPTIONS) $(USER_OPTIONS))
+    #"$(CONFIG_COMMAND)" @($(<[1]:T).rsp:O=FC:<=@":>=":E=-L"$(LINKPATH)" -Wl,$(RPATH_OPTION)$(SPACE)-Wl,$(RPATH) -Wl,-rpath-link$(SPACE)-Wl,"$(RPATH_LINK)" -o "$(<:T)" $(START-GROUP) "$(>:T)" "$(LIBRARIES)" $(FINDLIBS-ST-PFX) -l$(FINDLIBS-ST) $(FINDLIBS-SA-PFX) -l$(FINDLIBS-SA) $(END-GROUP) $(OPTIONS) $(USER_OPTIONS))
 }
 
 actions link.dll bind LIBRARIES
