diff --git a/build/Jamfile.v2 b/build/Jamfile.v2
index fc591b2..f67fee3 100644
--- a/build/Jamfile.v2
+++ b/build/Jamfile.v2
@@ -33,7 +33,7 @@ for local v in NO_COMPRESSION
 
 if $(has_zlib)
 {
-    using zlib : : <build-name>boost_zlib <tag>@tag ;
+    using zlib : : <build-name>boost_zlib : <tag>@tag ;
     zlib-requirements =
         [ ac.check-library /zlib//zlib : <library>/zlib//zlib
           <source>zlib.cpp <source>gzip.cpp ] ;
@@ -53,7 +53,7 @@ else
 
 if $(has_bzip2)
 {
-    using bzip2 : : <build-name>boost_bzip2 <tag>@tag ;
+    using bzip2 : : <build-name>boost_bzip2 : <tag>@tag ;
     bzip2-requirements =
         [ ac.check-library /bzip2//bzip2 : <library>/bzip2//bzip2
           <source>bzip2.cpp ] ;
