diff --git a/include/boost/json/detail/charconv/detail/fast_float/bigint.hpp b/include/boost/json/detail/charconv/detail/fast_float/bigint.hpp
index fc29481..4af3323 100644
--- a/include/boost/json/detail/charconv/detail/fast_float/bigint.hpp
+++ b/include/boost/json/detail/charconv/detail/fast_float/bigint.hpp
@@ -9,11 +9,12 @@
 #define BOOST_JSON_DETAIL_CHARCONV_DETAIL_FASTFLOAT_BIGINT_HPP
 
 #include <boost/json/detail/charconv/detail/fast_float/float_common.hpp>
+#define string_rindex rindex
 #include <algorithm>
 #include <cstdint>
 #include <climits>
 #include <cstring>
-
+#undef rindex
 namespace boost { namespace json { namespace detail { namespace charconv { namespace detail { namespace fast_float {
 
 // the limb width: we want efficient multiplication of double the bits in
