diff --git a/include/boost/multiprecision/detail/standalone_config.hpp b/include/boost/multiprecision/detail/standalone_config.hpp
index e56859be..50e525a6 100644
--- a/include/boost/multiprecision/detail/standalone_config.hpp
+++ b/include/boost/multiprecision/detail/standalone_config.hpp
@@ -33,7 +33,8 @@
  #  endif
  #else
  #  if __cplusplus < 201402L
- #    warning "The minimum language standard to use Boost.Math will be C++14 starting in July 2023 (Boost 1.82 release)"
+// ignore this for now
+// #    warning "The minimum language standard to use Boost.Math will be C++14 starting in July 2023 (Boost 1.82 release)"
  #  endif
  #endif
 
