diff --git a/src/tools/vacpp.jam b/src/tools/vacpp.jam
index b6930447b..8cc52ee07 100644
--- a/src/tools/vacpp.jam
+++ b/src/tools/vacpp.jam
@@ -152,7 +152,8 @@ actions vacpp.link bind NEEDLIBS
 
 actions vacpp.link.dll bind NEEDLIBS
 {
-    xlC_r -G $(LINKFLAGS) -o "$(<[1])" $(HAVE_SONAME)-Wl,-soname$(SPACE)-Wl,$(<[-1]:D=) -L$(LIBPATH) -L$(STDLIBPATH) "$(>)" "$(NEEDLIBS)" "$(NEEDLIBS)" -l$(FINDLIBS) $(OPTIONS) $(USER_OPTIONS)
+    #xlC_r -G $(LINKFLAGS) -o "$(<[1])" $(HAVE_SONAME)-Wl,-soname$(SPACE)-Wl,$(<[-1]:D=) -L$(LIBPATH) -L$(STDLIBPATH) "$(>)" "$(NEEDLIBS)" "$(NEEDLIBS)" -l$(FINDLIBS) $(OPTIONS) $(USER_OPTIONS)
+    xlC_r -G $(LINKFLAGS) -o "$(<[1])" -o $(<[-1]:D=) -L$(LIBPATH) -L$(STDLIBPATH) "$(>)" "$(NEEDLIBS)" "$(NEEDLIBS)" -l$(FINDLIBS) $(OPTIONS) $(USER_OPTIONS)
 }
 
 actions vacpp.compile.c
